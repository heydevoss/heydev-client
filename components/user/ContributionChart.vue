<template>
    <vue-no-ssr>
        <div class="chart">
            
            <span class="title"> Contributions </span> 

            <br><chartjs-doughnut
                v-bind:bind="true"
                v-bind:datasets="datasets"
                v-bind:labels="labels"
                v-bind:option="options"
            />
        </div>
    </vue-no-ssr>
</template>

<script>
export default {
    data() {
        return {
            datasets: [
                {
                data: [ this.totalPullRequests, this.totalCommits, this.totalIssues],
                backgroundColor: ["#b388ff", "#2196f3", "#80d8ff"],
                hoverBackgroundColor: ["#673ab7", "#0000FF", "#00FFFF"]
                }
            ],
            labels: ["Pull Requests", "Commits", "Issues"],
            options: {
                legend: {
                    position: 'right'
                }
            }
        }
    },
    props: {
        totalPullRequests: Number,
        totalCommits: Number,
        totalIssues: Number
    }
};
</script>
<style>
.title{
    text-align: center;
    padding: 10px;
    font-size: 25px;
    color: black;
}

.chart{
    display: flex;
    flex-direction: column;
    align-items:baseline; 
    width: 500px;
    height: 500px;

}
</style>

