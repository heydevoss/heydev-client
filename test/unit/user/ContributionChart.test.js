import { shallowMount } from '@vue/test-utils'
import ContributionChart from '@/components/user/ContributionChart.vue'

describe('ContributionChart', () => {
  const contributor = {
    openPr: 20,
    closedPr: 2,
    openIssues: 1,
    closedIssues: 13,
    commits: 5
  }
  const factory = props =>
    shallowMount(ContributionChart, {
      propsData: {
        ...props
      },
      stubs: {
        'chartjs-doughnut': '<div class="chartjs-doughnut-test"></div>',
        'vue-no-ssr': true
      }
    })

  const wrapper = factory({
    openPrContributor: contributor.openPr,
    closedPrContributor: contributor.closedPr,
    openIssuesContributor: contributor.openIssues,
    closedIssuesContributor: contributor.closedIssues,
    commitsContributor: contributor.commits
  })

  test('should be a Vue instance and render correctly', () => {
    expect(wrapper.isVueInstance()).toBeTruthy()
    expect(wrapper.element).toMatchSnapshot()
  })

  it('should render the chartjs-bar', () => {
    expect(wrapper.html().includes('<div class="chartjs-doughnut-test"></div>'))
  })

  it('should contain contributor data', () => {
    expect(wrapper.vm.openPrContributor).toBe(contributor.openPr)
    expect(wrapper.vm.closedPrContributor).toBe(contributor.closedPr)
    expect(wrapper.vm.openIssuesContributor).toBe(contributor.openIssues)
    expect(wrapper.vm.closedIssuesContributor).toBe(contributor.closedIssues)
    expect(wrapper.vm.commitsContributor).toBe(contributor.commits)
  })

  it('should contain the correct labels', () => {
    expect(wrapper.vm.labels.includes('Pull Requests Open'))
    expect(wrapper.vm.labels.includes('Pull Requests Closed'))
    expect(wrapper.vm.labels.includes('Pull Requests Open'))
    expect(wrapper.vm.labels.includes('Issues Open'))
    expect(wrapper.vm.labels.includes('Issues Closed'))
    expect(wrapper.vm.labels.includes('Commits'))
  })
  it('should render when no props are passed', () => {
    const emptyWrapper = factory()
    expect(emptyWrapper.isVueInstance()).toBeTruthy()
  })
})
