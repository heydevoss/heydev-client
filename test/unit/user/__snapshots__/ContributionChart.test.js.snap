// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContributionChart should be a Vue instance and render correctly 1`] = `
<vue-no-ssr-stub>
  <div>
    <br />
     
    <div
      bind="true"
      class="chartjs-doughnut-test chart"
      datasets="[object Object]"
      labels="Pull Requests Open,Pull Requests Closed,Issues Open,Issues Closed,Commits"
      option="[object Object]"
    />
  </div>
</vue-no-ssr-stub>
`;
